nrf_wifi/hw_if/hal/inc/hal_mem.h

```diff
@@ -14,71 +10,85 @@
 
 #include "hal_api.h"
 
+/**
+ * Enum defining the types of RPU memory.
+ */
 enum HAL_RPU_MEM_TYPE {
+	/** GRAM memory type */
 	HAL_RPU_MEM_TYPE_GRAM,
+	/** PKTRAM memory type */
 	HAL_RPU_MEM_TYPE_PKTRAM,
+	/** Core ROM memory type */
 	HAL_RPU_MEM_TYPE_CORE_ROM,
+	/** Core RET memory type */
 	HAL_RPU_MEM_TYPE_CORE_RET,
+	/** Core SCRATCH memory type */
 	HAL_RPU_MEM_TYPE_CORE_SCRATCH,
+	/** Maximum number of memory types */
 	HAL_RPU_MEM_TYPE_MAX
 };
 
-
 /**
- * hal_rpu_mem_read() - Read from the RPU memory.
- * @hpriv: Pointer to HAL context.
- * @host_addr: Pointer to the host memory where the contents read from the RPU
- *             memory are to be copied.
- * @rpu_mem_addr: Absolute value of the RPU memory address from which the
- *                contents are to be read.
- * @len: The length (in bytes) of the contents to be read from the RPU memory.
+ * Read from the RPU memory.
```

Comment:
```suggestion
 * @brief Read from the RPU memory.
```
Make sure to have @brief for all short function descriptions.